phoenix-core/src/main/java/org/apache/phoenix/compile/DeleteCompiler.java

```diff
@@ -323,7 +324,8 @@ public MutationPlan compile(DeleteStatement delete) throws SQLException {
                 // - read-only VIEW 
                 // - transactional table with a connection having an SCN
                 // TODO: SchemaUtil.isReadOnly(PTable, connection)?
-                if ( table.getType() == PTableType.VIEW && table.getViewType().isReadOnly() ) {
+                if ( ( table.getType() == PTableType.VIEW && table.getViewType().isReadOnly() ) 
+                     || ( table.isTransactional() && connection.getSCN() != null ) ) {
```

Comment:
Let's create a new SQLExceptionCode and throw that rather than a ReadOnlyTableException for the case of a connection specifying an SCN for a transactional table.